doc/api-extensions.md

```diff
@@ -2281,3 +2281,6 @@ When enabled and a suitable system is in use (requires ZFS 2.2 or higher), the Z
 
 This introduces support for the `all-projects` query parameter for the GET API calls to both `/1.0/operations` and `/1.0/operations?recursion=1`.
 This parameter allows bypassing the project name filter.
+
+## `metadata_configuration`
+Adds `GET /1.0/metadata/configuration` API endpoint to retrieve the generated metadata configuration in a JSON format for all the cluster, instance and server options. The JSON structure adopts the structure ```"configs" > `ENTITY` > `ENTITY_SECTION` > "keys" > [<CONFIG_OPTION_0>, <CONFIG_OPTION_1>, ...]``` where `ENTITY` can be `cluster` (with its `ENTITY_SECTION` that can take the value `cluster`), `instance` (with its `ENTITY_SECTION` that can take the values `boot`, `cloud-init`, `migration`, `miscellaneous`, `nvidia`, `raw`, `resource-limits`, `security`, `snapshots`, `volatile`) or `server` (with its `ENTITY_SECTION` that can take the values `acme`, `candid-and-rbac`, `cluster`, `core`, `images`, `loki`, `miscellaneous`, `oidc`)
```

Comment:
This is way too specific - these are the entities and sections that we have right now, but we will add others.

Reply:
What do you think about 
```md
Adds `GET /1.0/metadata/configuration` API endpoint to retrieve the generated metadata configuration in a JSON format for all the cluster, instance and server options. The JSON structure adopts the structure ```"configs" > `ENTITY` > `ENTITY_SECTION` > "keys" > [<CONFIG_OPTION_0>, <CONFIG_OPTION_1>, ...]```
```

Reply:
"for all the cluster, instance and server options" - this is right now. We want to use it for everything eventually.
If you want to make it clear that we're not covering everything yet, you could link to https://documentation.ubuntu.com/lxd/en/latest/config-options/ which has a list of options that are currently treated like this.
But we will never remember to go back here and update the description when we add more options (plus, should any of these descriptions be updated at all?).